base/setup/lib/utils/arcname.c

```diff
@@ -194,6 +198,10 @@ ArcGetNextTokenU(
         return NULL; /* Path starts with '(' and is thus invalid */
 
     SpecifierLength = (p - ArcPath) * sizeof(WCHAR);
+    if (SpecifierLength > UNICODE_STRING_MAX_CHARS)
```

Comment:
Be careful, SpecifierLength is a number of bytes, but using UNICODE_STRING_MAX_CHARS is in number-of-characters unit. It would be better to just use UNICODE_STRING_MAX_BYTES, or MAXUSHORT.
EDIT: In the other function you used MAXUSHORT, so you can use it there too.

Reply:
Fixed